homeassistant/components/media_player/__init__.py

```diff
@@ -753,24 +753,28 @@ def volume_up(self):
         if self.volume_level < 1:
             self.set_volume_level(min(1, self.volume_level + .1))
 
+    @asyncio.coroutine
     def async_volume_up(self):
-        """Turn volume up for media player.
-
-        This method must be run in the event loop and returns a coroutine.
-        """
-        return self.hass.loop.run_in_executor(None, self.volume_up)
+        """Turn volume up for media player."""
+        try:
+            yield from self.hass.loop.run_in_executor(None, self.volume_up)
+        except NotImplementedError:
+            yield from self.async_set_volume_level(
```

Comment:
I think we should get rid of the "helper" functionality, and have the ABC `volume_up/down` raise `NotImplementedError` unless they are overwritten. They'll behave the same as other functions and are only valid if the underlying library supports direct volume stepping.